scaffolding-python2/lib/scaffolding.sh

```diff
@@ -14,56 +14,127 @@
+    return 0
+  }
+}
+
 # The default_begin phase is executed prior to loading the scaffolding. As such
 # we have a scaffolding specific callback to allow us to run anything we need
 # to execute before download and build. This callback executes immediately
 # after the scaffolding is loaded.
 _scaffolding_begin() {
   pkg_deps=(${pkg_deps[@]} $scaffolding_python2_pkg)
+  pkg_build_deps=(${pkg_build_deps[@]} $scaffolding_python2_pkg)
+  pkg_env_sep=(
+    ['PYTHONPATH']=':'
+  )
+  # Disable download, verify, and unpack if we are only installing via pip.
+  # It should be noted that we currently must have `pkg_source` since we
+  # validate plans before scaffolding is loaded.
+  if [[ "$pkg_source" == "pip" ]] || [[ "$pkg_source" =~ ^nosuchfile.* ]] ; then
+    _pip_install_only
+  fi
 }
 
-
-# Recursively clean GOPATH and dependencies.
-do_default_clean() {
-  scaffolding_python2_clean
-}
-
+# Removes bytecode and other stuff we do not need to package up.
 scaffolding_python2_clean() {
-  pushd "$pkg_dirname"
-  if [[ -f "setup.py" ]]; then
+  if [[ $(_setup_py_exists) ]]; then
+    pushd "$HAB_CACHE_SRC_PATH/$pkg_dirname" >/dev/null
     python setup.py clean --all
+    popd >/dev/null
   else
-    find . \( -type f -name "*.pyc" -or -name "*.pyo" \) -delete
+    find "$HAB_CACHE_SRC_PATH/$pkg_dirname" \( -type f -name "*.pyc" -or -name "*.pyo" \) -delete
   fi
-  popd
 }
 
+# Run setup.py if it exists, otherwise return so pip can do the install.
 scaffolding_python2_build() {
-  python setup.py build
+  if [[ $(_setup_py_exists) ]]; then
+    build_line "Running setup.py"
+    python setup.py build
+  else
+    return 0
+  fi
 }
 
+# This can be called directly to install multiple pip packages.
 scaffolding_python2_pip_install() {
   local name
   local version
+  local args
   name="$1"
   version="$2"
+  args="${3:-"$name==$version --prefix=$pkg_prefix --force-reinstall --no-compile"}"
+
+  if [[ -n "$DEBUG" ]]; then
+    args="$args -vvv"
+  else
+    args="$args --quiet"
+  fi
 
-  pip install "$name==$version"
+  build_line "Running pip install"
+  # Disable shellcheck rule to ensure the pip install command is expanded
+  # with the correct syntax.
+  # shellcheck disable=SC2086
+  pip install $args
 }
 
+
+# Attempt to install the python package via setup.py without egg files.
+# We do not need the eggs since we are already keeping everything contained
+# within the hab package.
 scaffolding_python2_install() {
-  if [[ -f "${pkg_dirname}/setup.py" ]]; then
+  if [[ -n "$scaffolding_python2_pip_only" ]]; then
+    scaffolding_python2_pip_install "$pkg_scaffolding_python2_pip_name" "$pkg_version"
+  else
+    python setup.py install \
+      --prefix="$pkg_prefix" \
+      --old-and-unmanageable || \
+    python setup.py install \
+      --prefix="$pkg_prefix" \
+      --old-and-unmanageable || \
     python setup.py install \
       --prefix="$pkg_prefix"
   fi
+
+  add_path_env 'PYTHONPATH' 'lib/python3.6/site-packages'
```

Comment:
I am hard-coding these version here for the moment since we don't have an existing pattern to pull the package version from a dependency.